src/lib/Libattr/attr_fn_str.c

```diff
@@ -308,3 +308,29 @@ decode_jobname(attribute *patr, char *name, char *rescn, char *val)
 	}
 	return (decode_str(patr, name, rescn, val));
 }
+
+/**
+ * @brief
+ * 	set_str_extnd - set attribute value based upon another
+ *
+ *	A=B --> A is replaced with B
+ *
+ * @param[in]   attr - pointer to new attribute to be set (A)
+ * @param[in]   new  - pointer to attribute (B)
+ * @param[in]   op   - operator
+ *
+ * @return      int
+ * @retval      0       if ok
+ * @retval     >0       if error
+ *
+ */
+
+int
+set_str_extnd(struct attribute *attr, struct attribute *new, enum batch_op op)
+{
```

Comment:
The initial idea of us was that if an attribute does not support operations like += or -= we thought we will disallow them at least when adding new attributes from now onwards.  But as you said since this is a problem with our string resources as a whole, may be it is time to have architectural decision on changing set_str(). 

As suggested I am using set_str() only for now.